File: src/main/java/com/baulsupp/oksocial/util/LoggingUtil.java
package com.baulsupp.oksocial.util;

import com.google.common.collect.Lists;
import java.util.List;
import java.util.logging.ConsoleHandler;
import java.util.logging.Level;
import java.util.logging.LogManager;
import java.util.logging.LogRecord;
import java.util.logging.Logger;
import java.util.logging.SimpleFormatter;
import okhttp3.internal.http2.Http2;

public class LoggingUtil {
  private static List<Logger> activeLoggers = Lists.newArrayList();

  public static void configureLogging(boolean debug, boolean showHttp2Frames) {
    if (debug || showHttp2Frames) {
      LogManager.getLogManager().reset();
      ConsoleHandler handler = new ConsoleHandler();

      if (debug) {
        handler.setLevel(Level.ALL);
        handler.setFormatter(new OneLineLogFormat());
        Logger activeLogger = getLogger("");
        activeLogger.addHandler(handler);
        activeLogger.setLevel(Level.ALL);

        Logger x = getLogger("org.zeroturnaround.exec.stream");
        x.setLevel(Level.INFO);
      } else if (showHttp2Frames) {
        Logger activeLogger = getLogger(Http2.class.getName());
        activeLogger.setLevel(Level.FINE);
        handler.setLevel(Level.FINE);
        handler.setFormatter(new SimpleFormatter() {
          @Override public String format(LogRecord record) {
            return String.format("%s%n", record.getMessage());
          }
        });
        activeLogger.addHandler(handler);
      }
    }
  }

  public static Logger getLogger(String name) {
    Logger logger = Logger.getLogger(name);
    activeLoggers.add(logger);
    return logger;
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/util/LoggingUtil.kt
package com.baulsupp.oksocial.util

import com.google.common.collect.Lists
import io.netty.util.internal.logging.InternalLoggerFactory
import io.netty.util.internal.logging.JdkLoggerFactory
import okhttp3.internal.http2.Http2
import java.util.logging.ConsoleHandler
import java.util.logging.Level
import java.util.logging.LogManager
import java.util.logging.LogRecord
import java.util.logging.Logger
import java.util.logging.SimpleFormatter

class LoggingUtil {
  companion object {
    private val activeLoggers = Lists.newArrayList<Logger>()

    fun configureLogging(debug: Boolean, showHttp2Frames: Boolean) {
      InternalLoggerFactory.setDefaultFactory(JdkLoggerFactory.INSTANCE)

      if (debug || showHttp2Frames) {
        LogManager.getLogManager().reset()
        val handler = ConsoleHandler()

        if (debug) {
          handler.level = Level.ALL
          handler.formatter = OneLineLogFormat()
          val activeLogger = getLogger("")
          activeLogger.addHandler(handler)
          activeLogger.level = Level.ALL

          getLogger("org.zeroturnaround.exec").level = Level.INFO
          getLogger("io.netty").level = Level.INFO
          getLogger("io.netty.resolver.dns").level = Level.FINE
        } else if (showHttp2Frames) {
          val activeLogger = getLogger(Http2::class.java.name)
          activeLogger.level = Level.FINE
          handler.level = Level.FINE
          handler.formatter = object : SimpleFormatter() {
            override fun format(record: LogRecord): String {
              return String.format("%s%n", record.message)
            }
          }
          activeLogger.addHandler(handler)
          getLogger("io.netty.resolver.dns.DnsServerAddresses").level = Level.SEVERE
        }
      } else {
        getLogger("io.netty.resolver.dns.DnsServerAddresses").level = Level.SEVERE
      }
    }

    fun getLogger(name: String): Logger {
      val logger = Logger.getLogger(name)
      activeLoggers.add(logger)
      return logger
    }
  }
}
